routes/error-tracker.js

```diff
@@ -40,19 +40,57 @@ const GAE_METADATA = {
   severity: 500, // Error.
 };
 
+/** Extract parameters from a GET or POST request. */
+function getRequestParams(req) {
+  return req.method === 'POST' ? req.body : req.query;
+}
+
 /** Logs an event to Stackdriver. */
-function logEvent(log, event) {
-  return new Promise((resolve, reject) => {
-    log.write(log.entry(GAE_METADATA, event), writeErr => {
-      if (writeErr) {
-        reject(writeErr);
-      } else {
-        resolve();
-      }
-    });
-  });
+async function logEvent(log, event) {
+  await log.write(log.entry(GAE_METADATA, event));
 }
 
+/**
+ * Construct an event object for logging.
+ * @param {!Request} req
+ * @param {!Object<string, string|function>} reportingParams
+ * @param {!LogTarget} logTarget
+ * @return {Promise<?Object<string, string>>} event object, or `null` if the
+ *    error should be ignored.
+ */
+async function buildEvent(req, reportingParams, logTarget) {
+  const { buildQueryString, message, stacktrace, version } = reportingParams;
+
+  const stack = standardizeStackTrace(stacktrace, message);
+  if (ignoreMessageOrException(message, stack)) {
+    return null;
+  }
+  const unminifiedStack = await unminify(stack, version);
```

Comment:
This could fail in the old path, and we caught it and `console.error`d.

Reply:
Done. In the old path, we were sending a `202: Accepted before trying to unminify. Do you think that's still the right status, or would a 500 be more appropriate to indicate an error?

Reply:
A 500 would be appropriate.

Reply:
Done.